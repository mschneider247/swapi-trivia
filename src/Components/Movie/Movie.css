.movie-card {
  background-color: #535353;

}
