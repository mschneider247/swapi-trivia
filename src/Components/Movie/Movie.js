import React from 'react';
import './Movie.css';
import { Component } from 'react';
import { Link } from 'react-router-dom';
import { getCharacterData } from '../../apiCalls/apiCalls';

class Movie extends Component {
  constructor(props) {
    super()
    this.state = {
      characters: [],
      waitingForLoad: true
    }
    this.props = props;
    this.movieData = {
      episode_id : this.props.episode_id,
      title: this.props.title,
      opening_crawl: this.props.opening_crawl
    }
  }

  componentDidMount = () => {
    console.log("Movie component Did Mount, Now Fetching Character Data")
    let fetchedCharacters = [];
    this.props.characters.forEach((charac) => {
      getCharacterData(charac)
        .then(character => {
          return {
            name: character.name,
            homeworld: character.res[0].home,
            population: character.res[0].population,
            species: character.res[1],
            films: character.res[2][0],
            url: charac
          }
        })
        .then(character => {
          fetchedCharacters.push(character)
          if (fetchedCharacters.length === 10) {
            setInterval(() => {
              this.setState({ waitingForLoad: false}) 
            }, 500);
          }
        })
        .catch(error => console.error('error'))
    })
    this.setState({ 
      characters: fetchedCharacters 
    })
  }

  render() {
    return (
      <section>
        <div className="movie-card">
          <h3>{this.props.title}</h3>
          <h4><span className="bold-text">EPISODE</span> {this.props.episode_id}</h4>
          <h4><span className="bold-text">RELEASE</span> {this.props.release_date}</h4>
        </div>
        <div className="view-charac">
          {this.state.waitingForLoad ?
          <img src='https://media3.giphy.com/media/hQgJCEdGOEHa8/source.gif' alt="loading..."/> : <Link className="link" to={`movies/${this.props.episode_id}`}><h5 onClick={() => this.props.updateCurrentCharacters(this.state.characters, this.movieData)}>VIEW CHARACTERS</h5></Link>}
        </div>
      </section>
    )
}
}

export default Movie;
