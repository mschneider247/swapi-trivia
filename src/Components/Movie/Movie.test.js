import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Movie from './Movie';

describe('Movie', () => {
  it('should match snapshot with correct data', () => {
    const characters = [{name: "jim"}, {name: "nancy"}, {name: "stewart"}]
    const wrapper = shallow(
      <Movie
      title='Empire Strikes Back'
      episode_id='4'
      release_date='1998'
      id='4'
      key='4'
      characters={characters}/>
    );

    expect(wrapper).toMatchSnapshot()
  });

});
