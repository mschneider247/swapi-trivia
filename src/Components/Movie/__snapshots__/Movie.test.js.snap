// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match snapshot with correct data 1`] = `
<section>
  <div
    className="movie-card"
  >
    <h3>
      Empire Strikes Back
    </h3>
    <h4>
      <span
        className="bold-text"
      >
        EPISODE
      </span>
       
      4
    </h4>
    <h4>
      <span
        className="bold-text"
      >
        RELEASE
      </span>
       
      1998
    </h4>
  </div>
  <div
    className="view-charac"
  >
    <img
      alt="loading..."
      src="https://media3.giphy.com/media/hQgJCEdGOEHa8/source.gif"
    />
  </div>
</section>
`;
